### DISCUSSION TITLE
ResolveService for SRV Records and Active Directory

### DISCUSSION THREAD

**Discussion by mlippold** [MichaCo/DnsClient.NET#97] [General]
I'm trying to see if I can use the ResolveService to resolve Active Directory site specific records in DNS.  I've already have it resolving for broad "kerberos" and "ldap" records but I'd like to be able to be a bit more granular and only match records for a given Active Directory site.  
Some of the DNS SRV records look like this for AD site "us-midwest-il-nb" :
_kerberos._tcp.us-midwest-il-nb._sites
_kerberos._tcp.us-midwest-il-nb._sites.dc._msdcs
_ldap._tcp.us-midwest-il-nb._sites
_ldap._tcp.us-midwest-il-nb._sites.dc._msdcs

Is this possible?  If so, how would I make this look in my code?

Dim result As ServiceHostEntry() = client.ResolveService("myaddomain.com", "ldap", System.Net.Sockets.ProtocolType.Tcp)
For Each res In result
    Console.WriteLine(res.HostName)
Next

**Comment by MichaConrad** [comment]
Not really sure I understand what you are trying to do and what your problem is. Do you get an error, did you try some combinations of requests, the full address or rest address as prefix?

**Reply by mlippold** [reply]
No errors, just no records being returned.  I've tried all combinations of these (removing the lead underscores, all underscores, removing the ._tcp. portion etc):
_kerberos._tcp.us-midwest-il-nb._sites
_kerberos._tcp.us-midwest-il-nb._sites.dc._msdcs
_ldap._tcp.us-midwest-il-nb._sites
_ldap._tcp.us-midwest-il-nb._sites.dc._msdcs

I was plugging all of those attempted values into the "ldap" field you see below:
Dim result As ServiceHostEntry() = client.ResolveService("myaddomain.com", "ldap", System.Net.Sockets.ProtocolType.Tcp)

The only things that do return records are just the "kerberos" and "ldap" options - They are matching other DNS names out there:
_kerberos._tcp
_ldap._tcp

What I'd like to do is match on one of the types above so that I can just return DNS entries that pertain to my Active Directory site "us-midwest-il-nb" (having to specify "kerberos._tcp.us-midwest-il-nb" or "ldap._tcp.us-midwest-il-nb" wouldn't be a problem either).

It isn't the end of the world if I can't do this but I was hoping there was a way to only return records from a given Active Directory site. 

Love the library btw.

**Reply by MichaConrad** [reply]
@mlippold this depends what your DNS server you use responds. Try other command line tools and see if you get results for valid SRV requests, e.g. via `dig`

Instead of using the `ResolveService` extension, you can also just to the SRV queries yourself and see what those return, should be the same as what you get via nslookup or dig if all use the same DNS server
